lib/WUI/sntp/sntp_client.c

```diff
@@ -1,30 +1,50 @@
 #include "sntp.h"
 #include "sntp_client.h"
 #include "netdev.h"
+#include "netif.h"
+#include "netdb.h"
 
-static ip_addr_t ntp_server; // testing ntp server located in Prague
 static uint32_t sntp_running = 0; // describes if sntp is currently running or not
-void sntp_client_init(void) {
+void sntp_client_static_init(const char *ntp_address) {
     sntp_setoperatingmode(SNTP_OPMODE_POLL);
+    sntp_servermode_dhcp(0);
+    sntp_setservername(0, ntp_address);
+    sntp_init();
+}
 
-    /* TODO: enable DNS for ntp.pool.org as default sntp server*/
-
-    // TMP: ip of Czech CESNET NTP server tak.cesnet.cz
-    if (ipaddr_aton("195.113.144.238", &ntp_server)) {
-        sntp_setserver(0, &ntp_server);
-    }
+void sntp_client_dhcp_init(const char *ntp_address) {
+    sntp_setoperatingmode(SNTP_OPMODE_POLL);
+    sntp_servermode_dhcp(1);
+#if LWIP_IPV4
+    #ifdef SNTP_SERVER_DNS
+    sntp_setservername(1, "pool.ntp.org");
+    #else
+    sntp_setservername(1, ntp_address);
+    #endif
+#endif /* LWIP_IPV4 */
```

Comment:
This looks… weird. Both the LWIP_IPV4 thing, but more so the SNTP_SERVER_DNS thing… so in that case we ignore the provided address and use something hardcoded? And why there's no such thing in the `static init` thing?

Reply:
I removed the LWIP_IPV4 define for now.